### ISSUE TITLE
Proposal: Remove support for JSON files, and support just .yarn.txt

### ISSUE THREAD

**Issue by desplesda** [thesecretlab/YarnSpinner#173]
The ANTLR parser doesn't support reading JSON files, because it parses both the Yarn script as well as the metadata for each node. 

I'd like to get people's thoughts on simply removing the JSON support from Yarn Spinner, as well as possibly removing saving as JSON from the Yarn editor. I'd do it right away, but I'd like to get a feeling for how popular JSON is before I straight-up remove a feature 😊

Let me know what you think in the replies!

**Comment by robertpfeiffer** [comment]
I don't have a dog in the race about what format YarnSpinner takes, but I would like Yarn to continue to support saving as .json.

**Comment by McJones** [comment]
TLDR; as a format JSON is poorly suited to Yarn, its libraries add maintenance overhead and result in issues we either cant fix or take time away from maintaining and improving the core of Yarn Spinner.

In general I have a dislike of JSON as a file format but so with that caveat I think we should remove JSON support, and my reasoning is below.

In my view JSON is poorly suited for this work; yarn files aren’t logically broken up into objects, they aren’t indexed by keys, they have conditionals and links which are hard to describe semantically in JSON.
Sure we get around this but we get around it by abusing the format of JSON, then we still after this need the additional step of compiling so what does having it in JSON bring?
We are adding extra complexity just to then perform the regular compiler steps we’d need without supporting JSON.

Additionally JSON libraries are additional complexity that needs to be maintained.
Issues #72 #81 #93 #116 #119 #135 and #141 are all issues directly related to using JSON, they are seven different issues that I found with no real search effort, who knows how many more there are?
Sure some of these relate directly to the specific JSON library used which means it can be solved by changing that but then that another part we still have to hook in, test, and maintain.
Even if we replace Newtonsoft with another it will still have its own quirks to be understood and handled, we should be reducing requirements not adding to them.

**Comment by radiatoryang** [comment]
I posted this in the Slack, but basically: I think Yarn Spinner should prioritize a non-technical writer user when possible, and I think this user would likely prefer the simplicity of .yarn.txt over .json, at least when working on a game in Unity / C#. So I'd vote YES to removing JSON support from Yarn Spinner.

I have more mixed feelings about removing it from the Yarn editor. There are users who use, for example, [bondage.js](https://github.com/jhayley/bondage.js/) in their apps, who probably prefer JSON for whatever dev culture reasons, and in theory the main editor should be more platform-agnostic. So I'd vote NO to removing JSON support from the main editor.

**Comment by desplesda** [comment]
The conversation in Slack appears to have resolved along these lines:

- It's a good idea to simplify Yarn Spinner by removing JSON.
- However, other runtimes use JSON, and it's not a great idea to remove JSON saving from Yarn. This means that we'll have cases where users save their work in JSON, and load it in Yarn Spinner, only to find that it won't work.

Therefore, I'm going to do this:

- Remove JSON from Yarn Spinner
- Add code to Yarn Spinner that detects if a JSON file is being loaded, and rejects it with a clear description of what's going on, so that they know to re-save it as Yarn text.

Thanks for the discussion, all!
